File: cpp/day2a.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string command;
    int value = 0;
    int pos = 0, depth = 0;
    
    while (cin >> command >> value) {
        if (command.compare("up") == 0)
            depth -= value;
        else if (command.compare("down") == 0)
            depth += value;
        else
            pos += value;
    }

    cout << pos * depth << endl;

    return 0;
}

File: publish/day2a.py
#!/usr/bin/python3

def solve(input_fname):
    lines = []
    pos = depth = 0
    with open(input_fname) as ifile:
        for line in ifile:
            lines.append(line.strip().split())
    for cmd, unit in lines:
        unit = int(unit)
        if cmd == "forward":
            pos += unit
        elif cmd == "up":
            depth -= unit
        elif cmd == "down":
            depth += unit
    return pos * depth


if __name__ == "__main__":
    print(solve("../inputs/day2-demo.in"))
